### PR TITLE
ffmpeg: enable vulkan and libdrm

### PR DESCRIPTION
**Pull request by derpmalicious**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
- [x] I built this PR locally for these architectures:
  - [x] x86_64-glibc
  - [x] x86_64-musl

### CODE REVIEW

**Comment by q66** [comment]
make vulkan and drm `build_options`, put them in `build_options_default`

for `vulkan`, add conditional `makedepends` on `vulkan-loader Vulkan-Headers`

for `drm`, add conditional `makedepends` on `libdrm-devel`

**Comment by ahesford** [comment]
Looks like `libdrm-devel` is pulled in by `harfbuzz-devel -> cairo-devel -> MesaLib-devel -> libdrm-devel` and both `Vulkan-Headers` and `vulkan-loader` is pulled in by `SDL2-devel` and `vulkan-loader` is pulled in by `SDL2-devel` and  Unless the harfbuzz or SDL2 dependency chain changes, these `makedepends` should always be present. We could defer adding conditional `makedepends` until such a change forces our hand.

**Comment by q66** [comment]
that doesn't matter, dependencies that are explicitly needed should always be listed explicitly

File: srcpkgs/ffmpeg/template [hunk 1]
```diff
@@ -23,12 +23,14 @@ makedepends="zlib-devel bzip2-devel freetype-devel alsa-lib-devel libXfixes-deve
  $(vopt_if vpx libvpx-devel) $(vopt_if aom libaom-devel)
  $(vopt_if nvenc nv-codec-headers) $(vopt_if sndio sndio-devel)
  $(vopt_if dav1d libdav1d-devel) $(vopt_if zimg zimg-devel)
- $(vopt_if webp libwebp-devel) $(vopt_if sofa libmysofa-devel)"
+ $(vopt_if webp libwebp-devel) $(vopt_if sofa libmysofa-devel)
+ $(vopt_if vulkan "vulkan-loader Vulkan-Headers")
```
Comment (q66) [inline srcpkgs/ffmpeg/template:27]: single quotes within vopt_if

**Pull request merged**